### PR TITLE
Miniprot index and align

### PR DESCRIPTION
**Pull request by muffato**
Miniprot is the latest aligner, which can align protein sequences against genomic sequences: https://github.com/lh3/miniprot

This PR adds support for the `index` and `align` modes. The modules are very close to the minimap2 ones (because the CLI is almost identical) and borrow the same interface, layout, nomenclature, etc.

I also found that one of the minimap2 output channel was wrong in meta.yaml

## PR checklist

Closes #XXX 

- [x] This comment contains a description of changes (with reason).
- [x] If you've fixed a bug or added code that should be tested, add tests!
- [x] If you've added a new tool - have you followed the module conventions in the [contribution docs](https://github.com/nf-core/modules/tree/master/.github/CONTRIBUTING.md)
- [/] If necessary, include test data in your PR.
- [x] Remove all TODO statements.
- [x] Emit the `versions.yml` file.
- [x] Follow the naming conventions.
- [x] Follow the parameters requirements.
- [x] Follow the input/output options guidelines.
- [x] Add a resource `label`
- [x] Use BioConda and BioContainers if possible to fulfil software requirements.
- Ensure that the test works with either Docker / Singularity. Conda CI tests can be quite flaky:
  - [x] `PROFILE=docker pytest --tag <MODULE> --symlink --keep-workflow-wd --git-aware`
  - [x] `PROFILE=singularity pytest --tag <MODULE> --symlink --keep-workflow-wd --git-aware`
  - [x] `PROFILE=conda pytest --tag <MODULE> --symlink --keep-workflow-wd --git-aware`

### CODE REVIEW

File: modules/nf-core/miniprot/align/main.nf [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+    label 'process_medium'
+
+    conda (params.enable_conda ? "bioconda::miniprot=0.5" : null)
+    container "${ workflow.containerEngine == 'singularity' && !task.ext.singularity_pull_docker_container ?
+        'https://depot.galaxyproject.org/singularity/miniprot:0.5--h7132678_0':
+        'quay.io/biocontainers/miniprot:0.5--h7132678_0' }"
+
+    input:
+    tuple val(meta), path(pep)
+    path ref
```
Comment (matthdsm) [inline modules/nf-core/miniprot/align/main.nf:12]: Can you add a meta to the index cfr most other aligners?

**Review by matthdsm — CHANGES_REQUESTED** [review]
One small comment!

File: modules/nf-core/miniprot/align/main.nf (hunk 1 above)
Comment (muffato) [reply modules/nf-core/miniprot/align/main.nf:12]: 👍🏼

**Review by matthdsm — APPROVED** [review]
Go for it!

**Pull request merged**